projects/plugins/jetpack/extensions/blocks/subscriptions/class-jetpack-subscription-site.php

```diff
@@ -88,6 +91,79 @@ protected function get_post_end_placement_block_attributes( $default_attrs, $anc
 		return $default_attrs;
 	}
 
+	/**
+	 * Returns true if context is recognized as a header element.
+	 *
+	 * @param WP_Block_Template|WP_Post|array $context The block template, template part, or pattern the anchor block belongs to.
+	 *
+	 * @return bool
+	 */
+	protected function is_header_context( $context ) {
+		if ( $context instanceof WP_Post && $context->post_type === 'wp_navigation' ) {
+			return true;
+		}
+
+		if ( $context instanceof WP_Block_Template && $context->area === 'header' ) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Handles Subscription block navigation placement.
+	 *
+	 * @return void
+	 */
+	protected function handle_subscribe_block_navigation_placement() {
+		global $wp_version;
+
+		$is_enabled = get_option( 'jetpack_subscriptions_subscribe_navigation_enabled', false );
+		if ( ! $is_enabled ) {
+			return;
+		}
+
+		if ( ! wp_is_block_theme() || version_compare( $wp_version, '6.5-beta2', '<' ) ) { // TODO Fallback for classic themes and wp core < 6.5-beta2.
+			return;
+		}
+
+		add_filter(
+			'hooked_block_types',
+			function ( $hooked_blocks, $relative_position, $anchor_block, $context ) {
+				if (
+					$anchor_block === 'core/navigation' &&
+					$relative_position === 'last_child' &&
+					self::is_header_context( $context )
+				) {
+					$hooked_blocks[] = 'jetpack/subscriptions';
+				}
+
+				return $hooked_blocks;
+			},
+			10,
+			4
+		);
+
+		add_filter(
+			'hooked_block_jetpack/subscriptions',
+			function ( $hooked_block, $hooked_block_type, $relative_position, $anchor_block ) {
+				$is_navigation_anchor_block = isset( $anchor_block['blockName'] ) && $anchor_block['blockName'] === 'core/navigation';
+
+				if ( $is_navigation_anchor_block ) {
+					$class_name = ( ! empty( $hooked_block['attrs'] ) && ! empty( $hooked_block['attrs']['className'] ) )
```

Comment:
Could be we don't actually need this defence to ensure there wasn't user-defined classname, because when they modify the block it no longer is "hooked"?

Reply:
It might be. It doesn't hurt to have it here but you can test if it's really necessary 😄